examples_tests/41.VisibilityBuffer/main.cpp

```diff
@@ -496,46 +520,62 @@ int main()
         //set new pipeline (with overriden FS and layout)
         mb->setPipeline(std::move(newPipeline));
         //optionally adjust push constant ranges, but at worst it'll just be specified too much because MTL uses all 128 bytes
-#endif
     }
 
-#ifdef VT_READY
-    core::smart_refctd_ptr<video::IGPUTexturePacker> gpuTexPackers[ETP_COUNT];
-    for (uint32_t i = 0u; i < ETP_COUNT; ++i)
-        gpuTexPackers[i] = core::make_smart_refctd_ptr<video::IGPUTexturePacker>(driver, texPackers[i].get());
+    auto gpuvt = core::make_smart_refctd_ptr<video::IGPUVirtualTexture>(driver, am, vt.get());
+
+//#ifdef FOR_COMPUTE
+    core::smart_refctd_ptr<asset::ICPUDescriptorSetLayout> ds0layout;
+    {
+        asset::ICPUSampler::SParams params;
+        params.AnisotropicFilter = 3u;
+        params.BorderColor = asset::ISampler::ETBC_FLOAT_OPAQUE_WHITE;
+        params.CompareEnable = false;
+        params.CompareFunc = asset::ISampler::ECO_NEVER;
+        params.LodBias = 0.f;
+        params.MaxLod = 10000.f;
+        params.MinLod = 0.f;
+        params.MaxFilter = asset::ISampler::ETF_LINEAR;
+        params.MinFilter = asset::ISampler::ETF_LINEAR;
+        //phys addr texture doesnt have mips anyway and page table is accessed with texelFetch only
+        params.MipmapMode = asset::ISampler::ESMM_NEAREST;
+        params.TextureWrapU = params.TextureWrapV = params.TextureWrapW = asset::ISampler::ETC_CLAMP_TO_EDGE;
+        auto sampler = core::make_smart_refctd_ptr<asset::ICPUSampler>(params);
+
+        std::array<core::smart_refctd_ptr<asset::ICPUSampler>,VT_COUNT> samplers;
+        std::fill(samplers.begin(), samplers.end(), sampler);
+
+        params.AnisotropicFilter = 0u;
+        params.MaxFilter = asset::ISampler::ETF_NEAREST;
+        params.MinFilter = asset::ISampler::ETF_NEAREST;
+        params.MipmapMode = asset::ISampler::ESMM_NEAREST;
+        auto samplerPgt = core::make_smart_refctd_ptr<asset::ICPUSampler>(params);
+
+        auto bindings = vt->getDSlayoutBindings(PGTAB_BINDING, PHYSICAL_STORAGE_VIEWS_BINDING);
+        (*bindings)[0].samplers = &samplerPgt;
+        (*bindings)[1].samplers = samplers.data();
+
+        ds0layout = core::make_smart_refctd_ptr<asset::ICPUDescriptorSetLayout>(bindings->data(), bindings->data()+bindings->size());
+    }
```

Comment:
I really don't like that the bindings given by the `vt` don't use immutable samplers